src/psyclone/domain/lfric/lfric_stencils.py

```diff
@@ -0,0 +1,649 @@
+                    if not arg.stencil.extent:
+                        self._kern_args.append(arg)
+
+    @staticmethod
+    def extent_value(arg):
+        '''
+        Returns the content of the stencil extent which may be a literal
+        value (a number) or a variable name. This function simplifies this
+        problem by returning a string in either case.
+
+        :param arg: the argument with which the stencil is associated.
+        :type arg: :py:class:`psyclone.dynamo0p3.DynKernelArgument`
+
+        :returns: the content of the stencil extent.
+        :rtype: str
+
+        '''
+        if arg.stencil.extent_arg.is_literal():
+            return arg.stencil.extent_arg.text
+        return arg.stencil.extent_arg.varname
+
+    @staticmethod
+    def stencil_unique_str(arg, context):
+        '''
+        Creates a unique identifier for a stencil. As a stencil
+        differs due to the function space it operates on, type of
+        stencil and extent of stencil, we concatenate these things together
+        to create a unique string.
+
+        :param arg: kernel argument with which stencil is associated.
+        :type arg: :py:class:`psyclone.dynamo0p3.DynKernelArgument`
+        :param str context: a context for this stencil (e.g. "size" or \
+                            "direction").
+
+        :returns: unique string identifying the stencil for this argument.
+        :rtype: str
+
+        :raises GenerationError: if an explicit stencil extent is found in \
+                                 the meta-data for the kernel argument.
+        '''
+        unique = context
+        unique += arg.function_space.mangled_name
+        unique += arg.descriptor.stencil['type']
+        if arg.descriptor.stencil['extent']:
+            raise GenerationError(
+                "Found a stencil with an extent specified in the metadata. "
+                "This is not coded for.")
+        unique += arg.stencil.extent_arg.text.lower()
+        if arg.descriptor.stencil['type'] == 'xory1d':
+            unique += arg.stencil.direction_arg.text.lower()
+        return unique
+
+    def map_name(self, arg):
+        '''
+        Creates and registers a name for the stencil map associated with the
+        supplied kernel argument.
+
+        :param arg: kernel argument with which the stencil is associated.
+        :type arg: :py:class:`psyclone.dynamo0p3.DynKernelArgument`
+
+        :returns: a valid unique map name for a stencil in the PSy layer.
+        :rtype: str
+        '''
+        root_name = arg.name + "_stencil_map"
+        unique = LFRicStencils.stencil_unique_str(arg, "map")
+        return self._symbol_table.find_or_create_tag(unique, root_name).name
+
+    @staticmethod
+    def dofmap_symbol(symtab, arg):
+        '''
+        Creates and registers a symbol for the stencil dofmap associated with
+        the supplied kernel argument.
+
+        :param symtab: symbol table that will contain (or already contains) \
```

Comment:
Please remove the `\` character and align "the symbol" in the line below with "symbol table".